detection-rules/attachment_pdf_link_to_dmg.yml

```diff
@@ -0,0 +1,65 @@
+name: "Attachment: PDF with link to dmg file download"
+description: |
+  This rule identifies PDF attachments that either link directly to a DMG file or to a ZIP archive with a DMG file, or an encrypted ZIP containing a DMG file. This technique has been observed delivering MetaStealer Malware. 
+references:
+  - "https://thehackernews.com/2023/09/beware-metastealer-malware-targets.html"
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(attachments,
+          .file_type == "pdf"
+          and any(file.explode(.),
+                  any(.scan.url.urls,
+  
+                      // url links to dmg or zip
+                      (
+                        strings.iends_with(.url, ".dmg") or strings.iends_with(.url, "zip")
+                      )
+  
+                      // and downloads a dmg or a zip
+                      and any(beta.linkanalysis(.).files_downloaded,
+                              (
+                                .file_extension == "dmg"
+                                or (
+                                  .file_extension in~ $file_extensions_common_archives
+  
+                                  // and the zip contains a dmg file
+                                  and any(file.explode(.),
+                                          any(file.explode(..), .file_extension =~ "dmg")
+  
+                                          // exif inspection if encrypted
+                                          or strings.ends_with(.scan.exiftool.zip_file_name, ".dmg")
```

Comment:
could do an "is encrypted" check before this but i don't think it would actually matter

Reply:
i could go either way. lmk if you want it added. I don't think it changes the functionality either way.